pkg/certificates/certificate.go

```diff
@@ -78,3 +76,7 @@ func MakeCertificate(obj kmeta.OwnerRefableAccessor, name string) *cmv1.Certific
 		},
 	}
 }
+
+func deploymentName(sinkName string) string {
+	return kmeta.ChildName(sinkName, "-deployment")
```

Comment:
I think on the "generic" certificate factory we could just pass in the name of the deployment to the `MakeCert` function